packages/vite/src/node/config.ts

```diff
@@ -536,7 +544,10 @@ export async function resolveConfig(
       }
     },
     worker: resolvedWorkerOptions,
-    spa: config.spa ?? true
+    appType:
+      config.appType ?? config?.server?.middlewareMode === 'ssr'
```

Comment:
This is for backward compat. We can break it directly for v3 thought

Reply:
I think they are only a handful of frameworks out there using `middlewareMode: 'ssr'`  now that it should be safe to break. Otherwise another option is to `log` a warning, but framework authors would likely fix it right away so that end-users don't see it.